Refactor: RnD consoles and servers

SAADf603 opened the pull request:
## Описание
Переботка работы консолей и серверов.

## Ссылка на предложение/Причина создания ПР
https://discord.com/channels/617003227182792704/755125334097133628/1042039818940588092
https://discord.com/channels/617003227182792704/755125334097133628/1042904290265673799
https://discord.com/channels/617003227182792704/755125334097133628/1046015847535231016
https://discord.com/channels/617003227182792704/755125334097133628/1046877959467843624

## Демонстрация изменений

## To do

- [x] Пароли к серверам

- [x] Проверка консолью целостность сервера и отключение от сервера если уничтожен.

- [x] Невозможность работать без подключения к серверу

- [x] Автосинхронизация данных при увеличении уровня технологий

- [ ] Автолинковка

- [ ] Возможность отвязывать сервер, если необходимо

- [ ] Возможность передавать данные между серверами(новое меню?)

- [ ] Ремап сателита через В2

- [ ] Тесты

SAADf603 committed:
first changes

SAADf603 committed:
still bad...

SAADf603 committed:
so

SAADf603 committed:
Merge branch 'ss220-space:master220' into RnD-consoles-rework

SAADf603 committed:
better way

Furrior commented on code/modules/research/rdconsole.dm:
```diff
@@ -155,15 +155,10 @@ won't update every console in existence) but it's more of a hassle to do. Also,
-		else
-			return
-	for(var/obj/machinery/r_n_d/server/target)
+	var/abletosync(list)
```
Я писал же тебе var/list/abletosync = list()

SAADf603 committed:
Merge branch 'ss220-space:master220' into RnD-consoles-rework

SAADf603 committed:
here we go

+- привела в порядок и настроила основную задумку. Теперь он синхронизирует данные после каждой их загрузки и проверяет сервер на целостность перед тем, как что-то напечатать, в противном случае рвёт соединение и обнуляет технологии. Так же нельзя что-то сделать если нет подключения к серверу.

SAADf603 commented:
Ошибки из-за того, что я отделила все сервера в мире от таргета, с которым работает консоль, но не поправила это дальше. Поправлю позже.

SAADf603 committed:
dry

Furrior commented on code/modules/research/rdconsole.dm:
```diff
@@ -152,6 +154,52 @@ won't update every console in existence) but it's more of a hassle to do. Also,
+/obj/machinery/computer/rdconsole/proc/sync_server(/obj/machinery/r_n_d/server/target)
+	var/list/abletosync
+	for(var/obj/machines/r_n_d/server/S in GLOB.servers)
+		abletosync = list()
```
Это должно быть перед фором

SAADf603 commented:
истекло по сроку/заморожено
